pkg/skaffold/config/util_test.go

```diff
@@ -0,0 +1,222 @@
+		InsecureRegistries: []string{"bad.io", "worse.io"},
+		LocalCluster:       util.BoolPtr(true),
+		DefaultRepo:        "my-private-registry",
+	}
+	sampleConfig2 := &ContextConfig{
+		Kubecontext:        "another_context",
+		InsecureRegistries: []string{"good.io", "better.io"},
+		LocalCluster:       util.BoolPtr(false),
+		DefaultRepo:        "my-public-registry",
+	}
+
+	tests := []struct {
+		name           string
+		kubecontext    string
+		cfg            *GlobalConfig
+		expectedConfig *ContextConfig
+	}{
+		{
+			name: "global config when kubecontext is empty",
+			cfg: &GlobalConfig{
+				Global: &ContextConfig{
+					InsecureRegistries: []string{"mediocre.io"},
+					LocalCluster:       util.BoolPtr(true),
+					DefaultRepo:        "my-private-registry",
+				},
+				ContextConfigs: []*ContextConfig{
+					{
+						Kubecontext: someKubeContext,
+						DefaultRepo: "value",
+					},
+				},
+			},
+			expectedConfig: &ContextConfig{
+				InsecureRegistries: []string{"mediocre.io"},
+				LocalCluster:       util.BoolPtr(true),
+				DefaultRepo:        "my-private-registry",
+			},
+		},
+		{
+			name:           "no global config and no kubecontext",
+			cfg:            &GlobalConfig{},
+			expectedConfig: &ContextConfig{},
+		},
+		{
+			name:        "config for unknown kubecontext",
+			kubecontext: someKubeContext,
+			cfg:         &GlobalConfig{},
+			expectedConfig: &ContextConfig{
+				Kubecontext: someKubeContext,
+			},
+		},
+		{
+			name:        "config for kubecontext when globals are empty",
+			kubecontext: someKubeContext,
+			cfg: &GlobalConfig{
+				ContextConfigs: []*ContextConfig{sampleConfig2, sampleConfig1},
+			},
+			expectedConfig: sampleConfig1,
+		},
+		{
+			name:        "config for kubecontext without merged values",
+			kubecontext: someKubeContext,
+			cfg: &GlobalConfig{
+				Global:         sampleConfig2,
+				ContextConfigs: []*ContextConfig{sampleConfig1},
+			},
+			expectedConfig: sampleConfig1,
+		},
+		{
+			name:        "config for kubecontext with merged values",
+			kubecontext: someKubeContext,
+			cfg: &GlobalConfig{
+				Global: sampleConfig2,
+				ContextConfigs: []*ContextConfig{
+					{
+						Kubecontext: someKubeContext,
+					},
+				},
+			},
+			expectedConfig: &ContextConfig{
+				Kubecontext:        someKubeContext,
+				InsecureRegistries: []string{"good.io", "better.io"},
+				LocalCluster:       util.BoolPtr(false),
+				DefaultRepo:        "my-public-registry",
+			},
+		},
+		{
+			name:        "config for unknown kubecontext with merged values",
+			kubecontext: someKubeContext,
+			cfg:         &GlobalConfig{Global: sampleConfig2},
+			expectedConfig: &ContextConfig{
+				Kubecontext:        someKubeContext,
+				InsecureRegistries: []string{"good.io", "better.io"},
+				LocalCluster:       util.BoolPtr(false),
+				DefaultRepo:        "my-public-registry",
+			},
+		},
+		/* todo(corneliusweig): this behavior can be enabled with `mergo.WithAppendSlice` -> clarify requirements
```

Comment:
is this something you can address right now? what was stopping you from including it before?

Reply:
The main reason for not including this here is that it is a behavior change. That's why I opened #2554 for clarification and #2590 to implement that behavior on top of this change-set.
I'll remove the commented test from this PR to clarify that distinction in the git history.